### DISCUSSION TITLE
How to implement discount code?

### DISCUSSION THREAD

**Discussion by ulziibat-n** [EvgeniyMukhamedjanov/liquid-ajax-cart#48] [Q&A]
Hello!
Thanks for the awesome plugin. 
I'm using this plugin for a custom Shopify theme.
I need to discount code (Coupon) field on the my-card.liquid file.
I couldn't find about coupon code in your documentation.
Can you provide any guidance or information on this?
Thanks again.

**Comment by EvgeniyMukhamedjanov** [comment]
Hey!
The discount code functionality isn't a part of the Liquid Ajax Cart library as this functionality mostly done by apps and there is no a recommended way to do this. If you don't want to use any app you can try to make Ajax requests to the /discount/{discount_code} url. Try to find some info about this endpoint on how to use it on shopify community website.

**Comment by ulziibat-n** [comment]
Thank you very much for the quick response.

**Comment by andrewwise** [comment]
Hey @EvgeniyMukhamedjanov,

As of summer editions 2025, discount codes are now a native feature of the Cart Ajax API.

Changelog: https://shopify.dev/changelog/cart-ajax-api-discounts-support-on-cartupdatejs

Docs: https://shopify.dev/docs/api/ajax/reference/cart#update-discounts-in-the-cart

Any chance of this getting added to the roadmap? :slightly_smiling_face: 

Thanks!

**Reply by EvgeniyMukhamedjanov** [reply]
Hey @andrewwise ,
Yeah, it is in my roadmap :-) 
For now if you need to build the discount field, you can just use the [liquidAjaxCart.update()](https://liquid-ajax-cart.js.org/v2/liquid-ajax-cart-update/) method to send the discount request manually :-)

**Reply by andrewwise** [reply]
Fantastic, thanks for the reply!

**Reply by thagxt** [reply]
Hey Evgeniy, when do you think more or less this feature is going to get implemented in Liquid Ajax Cart? Do you have any date planned?

**Reply by EvgeniyMukhamedjanov** [reply]
Hey @thagxt ,
It is almost ready, you can preview it on the Demo store under the development theme:
https://liquid-ajax-cart.myshopify.com?preview_theme_id=126807081148
The only thing left is the error message for the discount form. It will be live this week!

**Reply by thagxt** [reply]
Hey @EvgeniyMukhamedjanov,

Just following up on this. Has there been any update on this? The demo looked nearly complete back then, so I'm wondering what happened with the final steps.

Thanks.

**Reply by andrewwise** [reply]
Hey @EvgeniyMukhamedjanov ,

Any update on if/when this might get merged in? :slightly_smiling_face:

**Reply by EvgeniyMukhamedjanov** [reply]
Hey @andrewwise :-) ,
I'm currently re-working the lib as bunch of tech debt issues have accumulated, so probably it will be ready only in the v3.0.0.
So the only way for now is using the [liquidAjaxCart.update()](https://liquid-ajax-cart.js.org/v2/liquid-ajax-cart-update/)

**Reply by andrewwise** [reply]
Thanks for the update!

**Comment by thagxt** [comment]
Amazing, thanks a lot! I’ve been cross-checking with Horizon and noticed something. If you add a discount code, then enter another code that doesn’t exist, the valid code that was applied gets removed. On Horizon, the applied discount code stays in place if the next code is invalid.

<s>I’m not sure if you’ve looked into this or if it depends on the JavaScript part at all or it's more how the discount codes are set up in the admin area. But on Horizon, you can also stack multiple discount codes together. In your demo store, it only allows one of the codes listed in the placeholder to be applied.</s> That depends on how discount codes are setup in Shopify.  

<details>
<summary>🎬 Screen Recordings</summary>

https://github.com/user-attachments/assets/9d35c8df-53ff-47e4-9c34-122a7df93dc2

https://github.com/user-attachments/assets/fd5405e2-e249-402a-b3d9-72a634ea37be

<img width="658" alt="Screenshot 2025-07-08 at 16 58 39" src="https://github.com/user-attachments/assets/3e31c06f-f2e4-413e-a139-0d62403446c0" />

</details>
